src/journal-remote/journal-gatewayd.c

```diff
@@ -496,10 +588,16 @@ static int request_handler_entries(
 
         if (m->cursor)
                 r = sd_journal_seek_cursor(m->journal, m->cursor);
+        else if (m->since_set && m->n_skip >= 0)
+                r = sd_journal_seek_realtime_usec(m->journal, m->since);
         else if (m->n_skip >= 0)
                 r = sd_journal_seek_head(m->journal);
+        else if (m->until_set && m->n_skip < 0)
```

Comment:
I cannot understand the condition at all... Could you explain?

Reply:
I've kept the (useless) test `m->n_skip < 0` (useless since the previous branch tests `m->n_skip >= 0`) and added the condition that when `until` was set, we should go at `until` and move the cursor backwards.
Note that we're in branches where `since` is not set (we tested it earlier).